src/lib/InstantSearch.ts

```diff
@@ -471,6 +471,11 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
     }
   }
 
+  public onChange = () => {
```

Comment:
```suggestion
  public onStateChange = () => {
```

this makes sense for me :) but we probably could add an _ in front of it to make it "internal"?

Reply:
I agree for the name of the function it's better like this. We don't have the concept of `state` though but `uiState`. It would be more consistent to name it: `onUiStateChange`. 

Not sure about the internal, it doesn't matter at the moment. We already have multiple attributes that are exposed publicly but only used internally (`helper`, `mainHelper`, `mainIndex`, ...).

Reply:
This function answers to any change event sent by the helper, so it's not specifically coupled with a `uiState` change, but rather any change. Saying that it's a "UI state change" is a lie. Later, we can think about using this hook for other purposes. Hence, `onStateChange` is generic yet accurate enough IMO.